### PR TITLE
Added check_lib_version utility to scarb api

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
**Stack**:
- #1510
- #1508
- #1501 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
Will be later used in forge & cast

File: crates/scarb-api/src/lib.rs [hunk 1]
```diff
@@ -191,6 +192,24 @@ pub fn name_for_package(metadata: &Metadata, package: &PackageId) -> Result<Stri
     Ok(package.name.clone())
 }
 
+/// Checks if all packages have a version compatible with the specified requirement
```
Comment (piotmag769) [inline crates/scarb-api/src/lib.rs:195]:
```suggestion
/// Checks if a package have a version compatible with the specified requirement
```

File: crates/scarb-api/src/lib.rs [hunk 2]
```diff
@@ -191,6 +192,24 @@ pub fn name_for_package(metadata: &Metadata, package: &PackageId) -> Result<Stri
...
+pub fn is_required_package_version(
```
Comment (piotmag769) [inline crates/scarb-api/src/lib.rs:196]:
```suggestion
pub fn is_package_compatible(
```

File: crates/scarb-api/src/lib.rs [hunk 3]
```diff
@@ -280,6 +299,48 @@ mod tests {
...
+        manifest_path
+            .write_str(&formatdoc!(
+                r#"
+                [package]
+                name = "version_checker"
+                version = "0.1.0"
+
+                [[target.starknet-contract]]
+                sierra = true
+                casm = true
```
Comment (piotmag769) [inline crates/scarb-api/src/lib.rs:316]: [nit] - don't need that since we use our sierra-casm compiler

**Review by piotmag769 — APPROVED** [review]

File: crates/scarb-api/Cargo.toml [hunk 4]
```diff
@@ -14,4 +14,5 @@ serde.workspace = true
 serde_json.workspace = true
 thiserror.workspace = true
 which.workspace = true
+semver = "1.0.0"
```
Comment (piotmag769) [inline crates/scarb-api/Cargo.toml:17]: We should add that to a top lvl Cargo.toml and inherit it here instead

Comment (MaksymilianDemitraszek) [reply crates/scarb-api/Cargo.toml:17]: It is moved there in next pr, here we use it only here.

File: crates/scarb-api/src/lib.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/scarb-api/src/lib.rs:196]: Done

File: crates/scarb-api/src/lib.rs (hunk 3 above)
Comment (MaksymilianDemitraszek) [reply crates/scarb-api/src/lib.rs]: Done

**Pull request merged**